### PR TITLE
`perfect_power` handles all Rationals

### PR DESCRIPTION
**Pull request by smichr**
#### Brief description of what is fixed or changed

The following used to raise a ValueError:
```python
>>> perfect_power(0)
False
>>> perfect_power(-9)
False
>>> perfect_power(-27)
(-3, 2)
>>> perfect_power(S.Half**3)
(1/2, 3)
```
#### Release Notes

* ntheory
    * `perfect_power` now only raises a ValueError for non-Rational input

### CODE REVIEW

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -420,12 +420,26 @@ def perfect_power(n, candidates=None, big=True, factor=True):
...
+    >>> perfect_power(-4)
+    False
+    >>> perfect_power(-8)
+    (-2, 3)
+
+    Rationals are also recognized:
+
+    >>> perfect_power(Rational(1/2)**3)
+    (1/2, 3)
+    >>> perfect_power(Rational(-3/2)**3)
```
Comment (oscarbenjamin) [inline sympy/ntheory/factor_.py:440]: Should this be e.g. `Rational(-3, 2)`?

Comment (smichr) [reply sympy/ntheory/factor_.py:440]: denominators that are powers of 2 are exactly converted...but it is better practice to do as you show
```python
>>> Rational(1/4)
1/4
```

Comment (smichr) [inline sympy/ntheory/factor_.py:440]:
```suggestion
    >>> perfect_power(Rational(1, 2)**3)
    (1/2, 3)
    >>> perfect_power(Rational(-3, 2)**3)
```

File: sympy/ntheory/tests/test_factor_.py [hunk 2]
```diff
@@ -157,6 +157,12 @@ def test_perfect_power():
         m = perfect_power(t*3**d, big=False)
         assert m and m[1] == 2 or d == 1 or d == 3, (d, m)
 
+    # negatives and non-integer rationals
+    assert perfect_power(-4) is False
+    assert perfect_power(-8) == (-2, 3)
+    assert perfect_power(Rational(1/2)**3) == (S.Half, 3)
+    assert perfect_power(Rational(-3/2)**3) == (-3*S.Half, 3)
```
Comment (smichr) [inline sympy/ntheory/tests/test_factor_.py:164]:
```suggestion
    assert perfect_power(Rational(1, 2)**3) == (S.Half, 3)
    assert perfect_power(Rational(-3, 2)**3) == (-3*S.Half, 3)
```

**Pull request merged**